test/unit/CitationFormat.test.js

```diff
@@ -0,0 +1,180 @@
+      expect(citationData.volume).to.equal('Test Volume');
+    });
+
+    it('should set series from work', () => {
+      const citationData = CitationFormatter.getCitationData(work, work.editions[0]);
+      expect(citationData.series).to.equal('The modern library classics');
+    });
+
+    it('should set a source link from the edition', () => {
+      const citationData = CitationFormatter.getCitationData(work, work.editions[0]);
+      expect(citationData.sourceLink).to.equal('test_link');
+    });
+
+    it('should set government document status', () => {
+      const citationData = CitationFormatter.getCitationData(work, work.editions[0]);
+      expect(citationData.isGovernmentDoc).to.equal(false);
+    });
+  });
+
+  describe('getAgentsOfType()', () => {
+    const testAgents = [
+      { name: 'test1', roles: ['test', 'other_again'] },
+      { name: 'test2', roles: ['other'] },
+      { name: 'test3', roles: ['test', 'other'] },
+      { name: 'test4', roles: ['other_again'] },
+    ];
+
+    it('should extract agents of type without exclude list', () => {
+      const testArray = CitationFormatter.getAgentsOfType(testAgents, 'test');
+      expect(testArray.length).to.equal(2);
+      expect(testArray).to.deep.equal(['test1', 'test3']);
+    });
+
+    it('should extract agents matching role, but without matching exclude roles', () => {
+      const testArray = CitationFormatter.getAgentsOfType(testAgents, 'other', ['test']);
+      expect(testArray.length).to.equal(1);
+      expect(testArray).to.deep.equal(['test2']);
+    });
+
+    it('should return agents with roles only not matching the exclude list', () => {
+      const testArray = CitationFormatter.getAgentsOfType(testAgents, null, ['test', 'other']);
+      expect(testArray.length).to.equal(1);
+      expect(testArray).to.deep.equal(['test4']);
+    });
+
+    it('should return an empty list if no agents are provided', () => {
+      expect(CitationFormatter.getAgentsOfType(null, 'test').length).to.equal(0);
+    });
+  });
+
+  describe('setLinkFields', () => {
+    it('should return a link and date from suplied items', () => {
+      const testItems = [
+        { links: [{ url: 'url1', modified: 'today' }] },
+        { links: [{ url: 'url2', modified: 'yesterday' }] },
+      ];
+      const testLink = CitationFormatter.setLinkFields(testItems);
+      expect(testLink.link).to.equal('url1');
+      expect(testLink.link_date).to.equal('today');
+    });
+
+    it('should return null for link attribute if no items are present', () => {
+      const testLink = CitationFormatter.setLinkFields(null);
+      expect(testLink.link).to.equal(null);
+      expect(testLink.link_date).to.equal(null);
+    });
+  });
+
+  describe('isGovernmentReport()', () => {
+    it('should return true if government_report measurement is 1', () => {
+      const testMeasures = [
+        { quantity: 'testing', value: 0 },
```

Comment:
can you change this value to 1 so that we make sure the "government_document" value trumps all other values?  

(Alternately, can be a different test, but I don't think it's necessary)